File: Lecture05_RecyclerView/AfterLecture_Java/app/src/main/java/no/hiof/larseknu/playingwithrecyclerview/MovieListFragment.java
package no.hiof.larseknu.playingwithrecyclerview;


import android.os.Bundle;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.navigation.NavDirections;
import androidx.navigation.Navigation;
import androidx.recyclerview.widget.GridLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Toast;

import java.util.ArrayList;

import no.hiof.larseknu.playingwithrecyclerview.model.Movie;

import no.hiof.larseknu.playingwithrecyclerview.adapter.MovieRecyclerAdapter;


/**
 * A simple {@link Fragment} subclass.
 */
public class MovieListFragment extends Fragment {
    private ArrayList<Movie> movieList;
    private RecyclerView movieRecyclerView;

    public MovieListFragment() {
        // Required empty public constructor
    }


    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        // Inflate the layout for this fragment
        View view = inflater.inflate(R.layout.fragment_movie_list, container, false);

        movieList = Movie.getMovies();

        return view;
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);

        movieRecyclerView = view.findViewById(R.id.movieRecyclerView);

        movieRecyclerView.setAdapter(new MovieRecyclerAdapter(view.getContext(), movieList, new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                // Gets the position of the item that's clicked
                int position = movieRecyclerView.getChildAdapterPosition(view);

                // Gets the movie based on which item got clicked
                Movie clickedMovie = movieList.get(position);

                // Creates the navigation action, including the uid argument
                NavDirections action = MovieListFragmentDirections.actionMovieListFragmentToMovieDetailFragment(clickedMovie.getUid());

                // Calls the navigat action, taking us to the MovieDetailFragment
                Navigation.findNavController(view).navigate(action);

                // Creates a toast with the movie that got clicked
                Toast.makeText(view.getContext(), clickedMovie.getTitle() + " clicked", Toast.LENGTH_LONG).show();
            }
        }));

        // Sets the layoutmanager we want to use
        movieRecyclerView.setLayoutManager(new GridLayoutManager(view.getContext(), 3));
        //movieRecyclerView.layoutManager = GridLayoutManager(context, 2)
        //movieRecyclerView.layoutManager = GridLayoutManager(context, 2, GridLayoutManager.HORIZONTAL, false)
        //movieRecyclerView.layoutManager = StaggeredGridLayoutManager(2, StaggeredGridLayoutManager.HORIZONTAL)
        //movieRecyclerView.layoutManager = LinearLayoutManager(context, LinearLayoutManager.HORIZONTAL, false)
        //movieRecyclerView.layoutManager = LinearLayoutManager(context, LinearLayoutManager.VERTICAL, false)

    }
}

File: Lecture05_RecyclerView/AfterLecture_Kotlin/app/src/main/java/no/hiof/larseknu/playingwithrecyclerview/MovieListFragment.kt
package no.hiof.larseknu.playingwithrecyclerview


import android.content.Intent
import android.os.Bundle
import androidx.fragment.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Toast
import androidx.navigation.NavController
import androidx.navigation.fragment.findNavController
import androidx.recyclerview.widget.GridLayoutManager
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.StaggeredGridLayoutManager
import kotlinx.android.synthetic.main.activity_main.*
import kotlinx.android.synthetic.main.fragment_movie_list.*
import no.hiof.larseknu.playingwithrecyclerview.adapter.MovieAdapter
import no.hiof.larseknu.playingwithrecyclerview.model.Movie

class MovieListFragment : Fragment() {
    private var movieList : ArrayList<Movie> = Movie.getMovies()

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        // Inflate the layout for this fragment
        return inflater.inflate(R.layout.fragment_movie_list, container, false)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        setUpRecycleView()
    }

    private fun setUpRecycleView() {
        // Set our own adapter to be used in the RecycleView, and sends it the data and creates the OnClickListener
        // With the listener gets called when an item in the list is clicked
        movieRecyclerView.adapter = MovieAdapter(movieList,
            View.OnClickListener { view ->
                // Gets the position of the item that's clicked
                val position = movieRecyclerView.getChildAdapterPosition(view)

                // Gets the movie based on which item got clicked
                val clickedMovie = movieList[position]

                // Creates the navigation action, including the uid argument
                val action = MovieListFragmentDirections.actionMovieListToMovieDetailFragment(clickedMovie.uid)

                // Calls the navigat action, taking us to the MovieDetailFragment
                findNavController().navigate(action)

                // Creates a toast with the movie that got clicked
                Toast.makeText(view.context, clickedMovie.title + " clicked", Toast.LENGTH_LONG).show();
            })

        // Sets the layoutmanager we want to use
        movieRecyclerView.layoutManager = GridLayoutManager(context, 3)
        //movieRecyclerView.layoutManager = GridLayoutManager(context, 2)
        //movieRecyclerView.layoutManager = GridLayoutManager(context, 2, GridLayoutManager.HORIZONTAL, false)
        //movieRecyclerView.layoutManager = StaggeredGridLayoutManager(2, StaggeredGridLayoutManager.HORIZONTAL)
        //movieRecyclerView.layoutManager = LinearLayoutManager(context, LinearLayoutManager.HORIZONTAL, false)
        //movieRecyclerView.layoutManager = LinearLayoutManager(context, LinearLayoutManager.VERTICAL, false)
    }
}
